contrib/babelfishpg_tsql/src/rolecmds.c

```diff
@@ -881,9 +885,21 @@ user_id(PG_FUNCTION_ARGS)
         if (!user_name)
             PG_RETURN_NULL();
 
-        len = strlen(user_name);
-        while (len > 0 && isspace(user_name[len-1]))
-        user_name[--len] = '\0';
+	i = strlen(user_input);
+	while (i > 0 && isspace((unsigned char) user_input[i - 1]))
+	user_input[--i] = '\0';
+	/* Convert login to lower-case */
+	for (i = 0; user_input[i]; i++)
+	{
+		user_input[i] = tolower(user_input[i]);
+	}
+	if (strcmp(user_input, "public") == 0)
+	{
+		PG_RETURN_OID(1);
+	}
```

Comment:
* misleading comment - It should be user.
* Let's move this piece of code before getting `get_physical_user_name` so that `user_name` is not allocated with some memory unnecessarily or else pfree `user_name` before returning.